### PR TITLE
Fix howto docs

### PR DESCRIPTION
**Pull request by rkratky**
## Description

This is a follow up to #429 -- don't merge before it.

Improve how-to docs:
    
- Move intro to how-to landing page
- Make headings consistently use "How to"
- Remove (post-build) "How to" from TOC

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] N/A: New/changed keys in YAML format are documented.
- [ ] N/A: \(Optional\) Adds example YAML for new feature.
- [x] \(Optional\) Closes an open bug in Launchpad. Fix [FR-6139](https://warthogs.atlassian.net/browse/FR-6139)

[FR-6139]: https://warthogs.atlassian.net/browse/FR-6139?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: doc/Makefile
Comment (daniloegea) [inline doc/Makefile:62]: thought: are the repetitive "How to"s really required in the titles? I mean, the section itself is called "How-to guides"... if we could remove them this hack wouldn't be necessary. Anyway, apparently there is no easy way to adjust the title in the sidebar and this string replacement seems to work just fine.

**Review by daniloegea — COMMENTED** [review]
Thanks, it looks good to me (only check the last commit).

File: doc/howto.md
Comment (slyon) [inline doc/howto.md:23]: question: Is this "name" value referenced somewhere? Do we need it? Depending on source language I translate it from, it could be insulting :D So if anything, I'd prefer an english term.

**Review by slyon — CHANGES_REQUESTED** [review]
Thanks, the changes are looking mostly good to me. But we hit some build failures. Locally, `make run` hits an endless loop and I see a failure on the readthedocs CI, which might be related to `doc/howtos.sh` being incomplete. (Some of which might already be fixed in #429)

I also left some inline comments.

File: doc/howto.md
Comment (rkratky) [reply doc/howto.md:23]: OMG, I'm an idiot! :) This was for testing purposes only. I wouldn't dream of leaving it in the content intentionally. Thank you for catching it!

File: doc/Makefile
Comment (rkratky) [reply doc/Makefile]: > I mean, the section itself is called "How-to guides"... if we could remove them this hack wouldn't be necessary.

True, but to keep it copnsistent with other Diataxis docs, this is how we're advised to do it: have "How to" in the heading but not in navigation.

> apparently there is no easy way to adjust the title in the sidebar and this string replacement seems to work just fine

I'm not happy with this hack. It's brittle and asks for trouble down the road. But modifying the navig. titles only is not possible with Markdown sources :(

**Comment by rkratky** [comment]
@slyon This is finally building successfully on RTD, so I believe it's good to merge (after #429).

> Thanks, the changes are looking mostly good to me. But we hit some build failures. Locally, `make run` hits an endless loop and I see a failure on the readthedocs CI, which might be related to `doc/howtos.sh` being incomplete. (Some of which might already be fixed in #429)
> 
> I also left some inline comments.

Thanks. I was aware of those problems. Should be fixed now.

**Comment by slyon** [comment]
@rkratky Could you please rebase this branch on top of `main`, in order to resolve the merge conflicts?

**Comment by rkratky** [comment]
> @rkratky Could you please rebase this branch on top of `main`, in order to resolve the merge conflicts?

Done. Sorry for the mess :)

File: doc/Makefile
Comment (slyon) [inline doc/Makefile:63]: This seems duplicated with `howto.sh`, can we re-use the script here, so we don't define the same logic in two places?

**Review by slyon — COMMENTED** [review]
I guess we could squash both commits into "[Improve how-to docs](https://github.com/canonical/netplan/pull/430/commits/4a1d1de2886242b7a3cb5d43160d87d8873a3f13)" (or I can do a squash merge). The first commit seems to be a left-over from the rebase.

Also see my inline comments for a few final nitpicks before we can merge this.

File: doc/Makefile
Comment (rkratky) [reply doc/Makefile:63]: > The first commit seems to be a left-over from the rebase.

Right. I'll squash them.

> It's still here :)

Aargh. It came back wirth the rebase.

> This seems duplicated with howto.sh, can we re-use the script here, so we don't define the same logic in two places?

OK

Comment (rkratky) [reply doc/Makefile]: > This seems duplicated with `howto.sh`, can we re-use the script here, so we don't define the same logic in two places?

I replaced the `sed` command (and the `howtos.sh` script) with a bit of Python code added to `conf.py`. That way, it's called as a part of the Sphinx build both on local and on RTD, and it also works when `make run` (.e. `sphinx-autobuild`) is used.

File: doc/howto.md
Comment (rkratky) [reply doc/howto.md]: Fixed :)

**Review by slyon — APPROVED** [review]
Thanks a lot for cleaning it all up! :heart:
LGTM.

**Pull request merged**
